lib/ansible/modules/windows/win_product_facts.ps1

```diff
@@ -0,0 +1,48 @@
+#!powershell
+# This file is part of Ansible
+
+# Copyright: (c) 2017, Dag Wieers (dagwieers) <dag@wieers.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+Function Get-ProductKey {
+    # First try to find the product key from ACPI
+    $product_key = (Get-CimInstance -Class SoftwareLicensingService).OA3xOriginalProductKey
+
+    if (-not $product_key) {
+        # Else try to get it from the registry instead
+        $data = Get-ItemPropertyValue -Path "HKLM:\Software\Microsoft\Windows NT\CurrentVersion" -Name DigitalProductId
```

Comment:
What if it's not there or not in the format you're expecting?

Reply:
The if-it's-not-there is now covered. If the format is not as I expect the resulting key will be corrupted. I wouldn't expect this to be the case though.